docs/developers.md

```diff
@@ -1,116 +1,144 @@
 
-##### bind.yaml
-
-TODO
-
-##### unbind.yaml
+Running the `apb init` will create a APB directory structure shown below:
+```bash
+$ apb init my-apb
+
+$ tree my-apb
+my-apb/
+├── apb.yml
+├── Dockerfile
+├── playbooks/
+└── roles/
+```
+### Spec File
+The `apb init` will auto generate a spec file.  You must edit this spec file to fit your application. The following is the spec file for `etherpad-apb`.  More examples can be found in the [apb-examples repo](https://github.com/fusor/apb-examples)
+
+```yml
+name: apb/etherpad-apb
+description: Note taking web application
+bindable: true
+async: optional
+parameters:
+  - name: hostport
+    description: The host TCP port as the external endpoint
+    type: int
+    default: 9001
+  - name: db_user
+    description: Database User
+    type: string
+  - name: db_pass
+    description: Database Password
+    type: string
+  - name: db_name
+    description: Database Name
+    type: string
+  - name: db_host
+    description: Database service hostname/ip
+    default: mariadb
+    type: string
+  - name: db_port
+    description: Database service port
+    type: int
+    default: 3306
+```
 
-TODO
+##### Spec File Parameters
+`parameters` section of the specfile; `ParameterObject` array
+
+Field Name | Type | Required | Default | Description
+---|---|---|---|---
+name | string| yes |  | The name of the parameter.
+required| bool | no | true | Whether or not the parameter is required.  If `false` and no default is provided, will be omitted from the parameters passed to the APB.
+description | string | yes | | A human readable description of the parameter.
+type | string | yes | | Type of parameter. `bool`, `int`, `float`, `string`, are valid
+default | bool,int,float,string|  no | | An optional default value for the parameter.
+
+### Actions
+Next we'll need to create `actions` for our APB.  At a minimum, we'll need to create the `provision.yml` and `deprovision.yml` under the `playbooks` folder.
+
+The `provision.yml` may look something like this:
+```yml
+- name: my-apb application
+  hosts: localhost
+  gather_facts: false
+  connection: local
+  roles:
+  - role: ansible.kubernetes-modules
+    install_python_requirements: no
+  - role: my-apb-openshift
+    playbook_debug: false
+```
 
-### Example Dockerfile
+And a simple `deprovision.yml` may look like this.
+```yml
+- hosts: localhost
+  gather_facts: false
+  connection: local
+  tasks:
+  - name: Delete my-apb project
+    command: oc delete project my-apb
 ```
-FROM ansibleplaybookbundle/apb-base
 
-MAINTAINER Dylan Murray <dymurray@redhat.com>
+We will also need to create the Ansible roles as specified in the actions. The below directory structure shows what it can look like:
 
-ADD apb/actions /apb/actions
+```bash
+$ tree my-apb
```

Comment:
You can probably drop ```$ tree my-apb``` unless the intention is for the user to run it.  But, since the result is going to be the same as the output below, maybe it's not needed? What do you think?